share/change.py

```diff
@@ -193,7 +218,12 @@ def __init__(self, graph):
         self.__graph = graph
         self.__visted = set()
         self.__visiting = set()
-        self.__nodes = list(graph.nodes)
+
+        # visit identifiers first
+        self.__nodes = sorted(
```

Comment:
This should be handled properly in the existing logic. If it is not, resorting the entire list is not the correct way to fix it.